src/imap.rs

```diff
@@ -114,6 +115,24 @@ impl Default for ImapConfig {
     }
 }
 
+#[derive(Debug, Fail, Clone, Eq, PartialEq)]
+enum SelectError {
+    #[fail(display = "Could not obtain imap-session object.")]
+    NoSession,
+
+    #[fail(display = "Connection Lost or no connection established")]
+    ConnectionLost,
+
+    #[fail(display = "imap-close (to expunge messages) failed: {}", _0)]
+    CloseExpungeFailed(String),
```

Comment:
Mark parameter with `#[cause]` like this: https://github.com/deltachat/deltachat-core-rust/pull/913/commits/9b27b1fbd084c6c730e0c69485a226f0481ae794

Reply:
thanks